instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/history/MapboxHistoryTestRule.kt

```diff
@@ -32,23 +38,31 @@ import java.util.concurrent.CountDownLatch
  */
 class MapboxHistoryTestRule : TestWatcher() {
 
+    private val mapboxTestDirectoryName = "mapbox_test"
     private val directory: File =
         File(
             Environment.getExternalStoragePublicDirectory(
                 Environment.DIRECTORY_DOWNLOADS
             ),
-            "mapbox_test"
+            mapboxTestDirectoryName
         )
 
     lateinit var historyRecorder: MapboxHistoryRecorder
+    private val context: Context by lazy {
+        InstrumentationRegistry.getInstrumentation().targetContext
+    }
 
     override fun finished(description: Description) {
         val filePath = historyRecorder.fileDirectory()!!
         val file = File(filePath)
         file.walk().filterNot { it.isDirectory }.forEach {
-            val path = description.methodName + File.separator + it.name
-            val target = File(directory, path)
-            it.copyTo(target)
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                saveFileUsingMediaStore(it, context, description.methodName, it.name)
+            } else {
+                val path = description.methodName + File.separator + it.name
+                val target = File(directory, path)
+                it.copyTo(target)
+            }
```

Comment:
Does this brunching bring any value for tests? We added`<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"
        tools:ignore="ScopedStorage" />` in the test app and old approach to write files works just fine, doesn't it?

Reply:
I don't think so, although I'm not 100% sure. I can't run instrumentation tests on API 33 just yet: https://github.com/mapbox/mapbox-navigation-android/issues/6134
But judging by the docs, this permission won't be enough: https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions
And the new introduces permissions are not exactly what we want (it's not audio, video or images).

Reply:
OK, I managed to workaround the issues and now the tests are running so I can look into it deeper.

Reply:
I checked: it does not work if we don't change anything. Although the tests are green, the files are not saved in the "Downloads" folder. It does work with Media Store API.